lalrpop/src/lr1/state_graph.rs

```diff
@@ -29,21 +31,18 @@ impl StateGraph {
             // Successors of a node arise from:
             // - shifts (found in the `conflicts` and `tokens` maps)
             // - gotos (found in the `gotos` map)
-            graph.extend_with_edges(
-                state
-                    .shifts
-                    .iter()
-                    .map(|(terminal, &state)| (Symbol::Terminal(terminal.clone()), state))
-                    .chain(
-                        state
-                            .gotos
-                            .iter()
-                            .map(|(nt, &state)| (Symbol::Nonterminal(nt.clone()), state)),
-                    )
-                    .map(|(symbol, successor)| {
-                        (NodeIndex::new(i), NodeIndex::new(successor.0), symbol)
-                    }),
-            );
+            let shifts = state
+                .shifts
+                .iter()
+                .map(|(terminal, &state)| (Symbol::Terminal(terminal.clone()), state));
+            let gotos = state
+                .gotos
+                .iter()
+                .map(|(nt, &state)| (Symbol::Nonterminal(nt.clone()), state));
+            let it = shifts.chain(gotos).map(|(symbol, successor)| {
+                (NodeIndex::new(i), NodeIndex::new(successor.0), symbol)
+            });
```

Comment:
this is just code simplification?

Reply:
Yes, `shifts.chain(gotos).` part was the equivalent longer chain before.